### ISSUE TITLE
Use Vercel to host the app

### ISSUE THREAD

**Issue by xcv58** [conwnet/github1s#33]
Using the GitHub page to host the app has several drawbacks:

1. Not easy to preview the pending change
2. No existing CI/CD support
3. No way to have a preview version for PR

I suggest using https://vercel.com/ to host the app. This is a version I deployed:  https://test.xcv58.vercel.app/conwnet/github1s/blob/HEAD/src/vs/code/browser/workbench/workbench.ts

There are two drawbacks:

## Price

And the drawback is Vercel is not a free service but the price is cheap https://vercel.com/pricing. I paid around $3 per month and the majority part is custom domains.

And I can help to request a free or discounted account for this project.

![image](https://user-images.githubusercontent.com/503123/107468869-9f8d6700-6b1d-11eb-80c7-8fba130e146f.png)

## ~~CI/CD setup~~

~~I don't know whether Vercel supports the `toolchain` to build VSCode. I could reach to their customer support to check whether there is any similar setup/use-case.~~

~~But in the worst case, we could use CircleCI (free service) to build the artifact and integrate it with Vercel.~~

--------

Just for reference 

The steps to deploy under my Vercel account:

1. Run `yarn` & `yarn build`
2. `cd dist` and make a `vercel.json` file with the content:
```
{
  "routes": [{
      "src": "/static/(.*)",
      "dest": "/images/$1"
    },
    {
      "src": "/favicon.ico",
      "dest": "/favicon.ico"
    },
    {
      "src": "/manifest.json",
      "dest": "/manifest.json"
    },
    {
      "src": ".*",
      "dest": "/index.html"
    }
  ]
}
```
3. Install the Vercel CLI https://vercel.com/download
4. Run `vercel login` command to login
5. Run `vercel` to deploy.

**Comment by xcv58** [comment]
I have a demo for Vercel CI/CD setup: https://github.com/xcv58/github1s/pull/2

And for every PR, it will deploy on Vercel and have a temporary URL on the change:

https://github1s-git-vercel.xcv58.vercel.app/xcv58/tab-manager-v2

![image](https://user-images.githubusercontent.com/503123/107478454-57c30b80-6b2e-11eb-861b-2353cff7fd64.png)

**Comment by xcv58** [comment]
@conwnet if you set up a Vercel account and merge this change https://github.com/xcv58/github1s/pull/2/files
it would host on Vercel automatically. For each master commit, it would link to a default production domain. For every PR, it will deploy and assign a temp domain.

You could easily switch github1s.com to Vercel as well if you want https://vercel.com/docs/custom-domains

**Comment by hediet** [comment]
I think github pages is a very robust solution to host it long term. Even if maintainance of this project dies, github pages will continue to serve it. Also, you actually can setup CD for gh pages and automate the release process! 

I doubt vercel is continuing to host the project when you stop payments.
Vercel is nice though for PRs. Maybe a combination makes sense? Gh page for the master, vercel for development?

Also, @conwnet, you should try to setup github sponsors ;) It might cover all costs.

**Comment by xcv58** [comment]
@hediet AFAIK, Vercel is free without using custom domain. So no payment issue in long term. 

And we could reach to their team to ask whether they could sponsor this open source project before make the decision.

**Comment by conwnet** [comment]
I think we have a responsibility to ensure the services run stable, maybe we can use vercel for development first, and we can switch to Vercel then if they could sponsor this open source project...🤣

**Comment by happybeing** [comment]
Apologies for going a bit off topic but I'm building a decentralised Git Portal to run from static storage (so no server side code) and wondering if github1s requires server side code, and if so for what. Hosting on Vercel/ghpages I assume not, but want to confirm.

**Comment by xcv58** [comment]
> I think we have a responsibility to ensure the services run stable, maybe we can use vercel for development first, and we can switch to Vercel then if they could sponsor this open source project...🤣

Sounds good, could you please merge this PR https://github.com/conwnet/github1s/pull/50, and set up a Vercel account https://vercel.com/docs/git/vercel-for-github

The setting looks like this:

![image](https://user-images.githubusercontent.com/503123/107550814-59202280-6b86-11eb-8ebd-585c4d472991.png)

**Comment by xcv58** [comment]
> Apologies for going a bit off topic but I'm building a decentralised Git Portal to run from static storage (so no server side code) and wondering if github1s requires server side code, and if so for what. Hosting on Vercel/ghpages I assume not, but want to confirm.

It's a pure static web app.

**Comment by softmarshmallow** [comment]
Build on vercel thrown some issues which i fixed on this fork.

"command compile not found"

github.surf
https://github.com/bridgedxyz/github.surf

**Comment by xcv58** [comment]
> Build on vercel thrown some issues which i fixed on this fork.
> 
> "command compile not found"
> 
> github.surf
> https://github.com/bridgedxyz/github.surf

It’s fixed https://github.com/conwnet/github1s/commit/90ccc6ad7bda62d8f37e5ce1312635b71942f6e0

Please use the latest master

**Comment by Siddhant-K-code** [comment]
@xcv58 some of the Latest PRs getting their `preview failed`, can you pls look into it?

**Comment by xcv58** [comment]
> @xcv58 some of the Latest PRs getting their `preview failed`, can you pls look into it?

Of course, I suspect it has an incorrect configuration on the Vercel side. However, the Vercel dashboard is only accessible by @conwnet .

I'll try to make a PR that doesn't need any customization from Vercel.

**Comment by xcv58** [comment]
PR to fix the Vercel preview issue #164

**Comment by Siddhant-K-code** [comment]
I Guess, All Done Here. Can we close this issue?
